text/0000-regex-1.0.md

```diff
@@ -0,0 +1,984 @@
+
+  /// Returns an iterator of successive non-overlapping matches with capturing
+  /// group information in the text given.
+  pub fn captures_iter<'r, 't>(&'r self, text: &'t str) -> CapturesIter<'r, 't>;
+}
+```
+
+(N.B. The `captures` method can technically replace all uses of `find` and
+`is_match`, but is potentially slower. Namely, the API reflects a performance
+trade off: the more you ask for, the harder the regex engine has to work.)
+
+There is one additional, but idiosyncratic, search method:
+
+```rust
+impl Regex {
+  /// Returns the end location of a match if one exists in text.
+  ///
+  /// This may return a location preceding the end of a proper leftmost-first
+  /// match. In particular, it may return the location at which a match is
+  /// determined to exist. For example, matching `a+` against `aaaaa` will
+  /// return `1` while the end of the leftmost-first match is actually `5`.
+  ///
+  /// This has the same performance characteristics as `is_match`.
+  pub fn shortest_match(&self, text: &str) -> Option<usize>;
+}
+```
+
+And two methods for splitting:
+
+```rust
+impl Regex {
+  /// Returns an iterator of substrings of `text` delimited by a match of
+  /// this regular expression. Each element yielded by the iterator corresponds
+  /// to text that *isn't* matched by this regex.
+  pub fn split<'r, 't>(&'r self, text: &'t str) -> SplitsIter<'r, 't>;
+
+  /// Returns an iterator of at most `limit` substrings of `text` delimited by
+  /// a match of this regular expression. Each element yielded by the iterator
+  /// corresponds to text that *isn't* matched by this regex. The remainder of
+  /// `text` that is not split will be the last element yielded by the
+  /// iterator.
+  pub fn splitn<'r, 't>(&'r self, text: &'t str, limit: usize) -> SplitsNIter<'r, 't>;
+}
+```
+
+And three methods for replacement. Replacement is discussed in more detail in a
+subsequent section.
+
+```rust
+impl Regex {
+  /// Replaces matches of this regex in `text` with `rep`. If no matches were
+  /// found, then the given string is returned unchanged, otherwise a new
+  /// string is allocated.
+  ///
+  /// `replace` replaces the first match only. `replace_all` replaces all
+  /// matches. `replacen` replaces at most `limit` matches.
+  fn replace<'t, R: Replacer>(&self, text: &'t str, rep: R) -> Cow<'t, str>;
+  fn replace_all<'t, R: Replacer>(&self, text: &'t str, rep: R) -> Cow<'t, str>;
+  fn replacen<'t, R: Replacer>(&self, text: &'t str, limit: usize, rep: R) -> Cow<'t, str>;
+}
+```
+
+And lastly, three simple accessors:
+
+```rust
+impl Regex {
+  /// Returns the original pattern string.
+  pub fn as_str(&self) -> &str;
+
+  /// Returns an iterator over all capturing group in the pattern in the order
+  /// they were defined (by position of the leftmost parenthesis). The name of
+  /// the group is yielded if it has a name, otherwise None is yielded.
+  pub fn capture_names(&self) -> CaptureNamesIter;
```

Comment:
it would be good to document that the whole match is included as the first (unnamed) capture and therefore this always yields `None` as first element.